wire-ios-transport/Source/Background/BackgroundActivityFactory.swift

```diff
@@ -156,7 +156,7 @@ import WireUtilities
             let count = SafeValueForLogging(activities.count)
             if activities.remove(activity) != nil {
                 WireLogger.backgroundActivity.info(
-                    "End background activity: removed \(activity), \(count) others left.",
+                    "End background activity: removed \(activity), \(count.safeForLoggingDescription) others left.",
```

Comment:
Not a request for change.

Have you looked into `ExpressibleByStringLiteral`. I think it can be quite useful for logging APIs and auto redacting, as it allows us to do things like:

```swift
log("Deleted ID: \(id)") // logs -> "Deleted ID: ****"
```

The literal parts have to be known at compile time and we can automatically redact dynamic parts based on their type for example. You can also do things like:

```swift
log("Deleted ID: \(noRedact: id)") // logs -> "Deleted ID: 1234"
```

Anyway it is pretty powerful and can help force a compile time that we don't log things accidentally.

Reply:
good idea I'll create a ticket to improve this, probably for develop